From ce2ad7ebd6c0fa15316c642105a84b1f7a687757 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Mon, 31 Oct 2016 19:51:56 +0100
Subject: [PATCH] convert angle from degrees to radians for glm

This makes simple (i.e., not implemented as a shader) OpenGL transitions
work correctly again.

(cherry picked from commit 7445399af829c48e8c71eed66f132d96fa195c37)

Change-Id: I773f686089bce3611940743b1a7f5046093886e8
---
 slideshow/source/engine/OGLTrans/generic/OGLTrans_Operation.cxx | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/slideshow/source/engine/OGLTrans/generic/OGLTrans_Operation.cxx b/slideshow/source/engine/OGLTrans/generic/OGLTrans_Operation.cxx
index c61209f..97d8902 100644
--- a/slideshow/source/engine/OGLTrans/generic/OGLTrans_Operation.cxx
+++ b/slideshow/source/engine/OGLTrans/generic/OGLTrans_Operation.cxx
@@ -31,6 +31,8 @@
 #include <glm/gtc/matrix_transform.hpp>
 #include <glm/gtc/type_ptr.hpp>
 
+#include <basegfx/numeric/ftools.hxx>
+
 #include "OGLTrans_Operation.hxx"
 
 SRotate::SRotate(const glm::vec3& Axis, const glm::vec3& Origin,
@@ -38,7 +40,7 @@ SRotate::SRotate(const glm::vec3& Axis, const glm::vec3& Origin,
     Operation(bInter, T0, T1),
     axis(Axis),
     origin(Origin),
-    angle(Angle)
+    angle(basegfx::deg2rad(Angle))
 {
 }
 
@@ -55,7 +57,7 @@ RotateAndScaleDepthByWidth::RotateAndScaleDepthByWidth(const glm::vec3& Axis,
     Operation(bInter, T0, T1),
     axis(Axis),
     origin(Origin),
-    angle(Angle),
+    angle(basegfx::deg2rad(Angle)),
     scale(bScale)
 {
 }
@@ -65,7 +67,7 @@ RotateAndScaleDepthByHeight::RotateAndScaleDepthByHeight(const glm::vec3& Axis,
     Operation(bInter, T0, T1),
     axis(Axis),
     origin(Origin),
-    angle(Angle),
+    angle(basegfx::deg2rad(Angle)),
     scale(bScale)
 {
 }
-- 
2.9.3

